File: MPI/Lab/hello_f/hello.f90
PROGRAM hello


  INTEGER, PARAMETER:: dp=KIND(0.d0) 

  INTEGER ::  ierr, comm_size, comm_rank, length
  REAL(dp) :: RESULT
  CHARACTER (len=100) :: proc_name = "test"

  WRITE(*,*) "Hello World from process = ", comm_rank, " on processor ", proc_name

  RESULT = EXP(REAL(comm_rank, dp))

  WRITE(*,*) "Exp(", comm_rank, ") = ", RESULT 

  WRITE(*,*) "Number of mpi processes = ", comm_size

END PROGRAM hello

File: MPI/Lab/hello_c/hello.c

#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int main (int argc, char *argv[]) 
{
  int comm_size, comm_rank, length;
  char proc_name[100] = "test";
  double result;

  comm_rank = 0;
  comm_size = 1;
  
  printf("Hello World from process = %d on processor %s\n", comm_rank, proc_name);
  result = exp((double) comm_rank);
  printf("Exp(%d) = %f\n", comm_rank, result);

  printf("Number of mpi processes = %d\n", comm_size);
} 
